internal/cmdexport/orchestrators/jenkins.go

```diff
@@ -0,0 +1,160 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package orchestrators
+
+import (
+	"bytes"
+	"fmt"
+	"path"
+	"strings"
+	"text/template"
+
+	"github.com/GoogleContainerTools/kpt/internal/cmdexport/types"
+)
+
+type Jenkins struct {
+	Stage JenkinsStage
+}
+
+func (p *Jenkins) Init(config *types.PipelineConfig) Pipeline {
+	step := JenkinsStageStep{
+		MountedWorkspace: "/app",
+		Dir:              config.Dir,
+		FnPaths:          config.FnPaths,
+	}
+	p.Stage = JenkinsStage{
+		Name: "Run kpt functions",
+		Scripts: []string{
+			step.Generate(),
```

Comment:
Is that possible to assign the step and stages in `Init` function? Then in `Generate()`, we can call all underlying `Generate()` from step or stage?